### ISSUE TITLE
Simplify TXT support with a new opinion: TXT records should act like one big string.

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#1592]
[NOTE: This would be a breaking change. Therefore, it would have to wait for 4.0.]

### Problem: TXT record support is very complex for very little benefit.

The RFC define TXT records as having multiple strings, each of length 0 to 255 octets.  Some providers accept a list of strings per the RFCs.  However most providers only permit a single string in their API, but split it into 255-octet chunks for DNS service (i.e. the protocol on port 53).  Some just permit a single string length 1 to 255 octets.

DNSControl tries to support all of these variations.  This leads to a lot of code complexity.  There are exceptions for TXT records all over the code.

### Proposal: Remove support for user-specified multiple strings.  

Simple accept 1 string of any length, and quietly break it into smaller strings "behind the scenes" when the provider requires this.

DNSControl is an [opinionated](https://stackexchange.github.io/dnscontrol/opinions) system. For example, "Opinion #4: All DNS is lowercase" means we downcase any DNS labels.  We would add a new opinion: "TXT records should be treated as one long string; multiple substrings are an issue for the protocol's wire-format."

That is, the user would lose control over where strings are split.  We would still support multiple strings, but they would be invisible to the user.

If we receive a TXT record from a provider with multiple smaller strings, merge them into one long string.   When we send updates to the provider via the API, split the string as the provider requires.  Splitting will be done on 255-octet boundaries.

In other words, we would treat the substrings of a TXT record as building-blocks of a larger string.  I think this is what the designers of TXT records intended.

### What might break:  Users that depend on varying string lengths.

For example, normally we receive a multiple substrings that are all 255-octets except the for last one (the "remainder").  So, for example, if the full string is 600 octets long, the substrings will be of length 255, 255, and 90.  We will combine those substrings into a single 600-octet string internally.  If we send an update to the API, the string will be silently split into substrings of length 255, 255, 90 and the provider will be none the wiser.

However, what if we receive strings of length 200, 90, 55, and 255?   The next "dnscontrol push" could send an update reorganizing them into 255, 255, and 90. (the same total 600-octets, in the same order, just split differently).

### Why I think this is low-risk.

I don't know of any service or protocol that depends on TXT records being split a particular way.  If a new service did require this, half the DNS providers in the world wouldn't support it.

I don't know of any service or protocol that depends on TXT records being joined in a particular way.  As far as I can tell, everyone simply concatenates each substring.  None pay attention to the substrings as positional values, for example.  Again, if a new service did require this, half the DNS providers in the world wouldn't support it.

If someone did complain that their "200, 90, 55, and 255" substrings were reorganized into substrings of length "255, 255, and 90" ... well, I think we could make the case that they're asking for trouble.

### What would the change look like?

1. Remove the "multiple string" syntax for TXT records in dnsconfig.js.

CURRENT:  Syntax permits TXT records to be specified as such:

```
      TXT('single', '598611146-3338560'),   /// one string of any length
      TXT('multiple', ['one', 'two', 'three']),  // Multiple strings
```

PROPOSED:  Remove the "multiple strings" option.

In 3.x such syntax would be permitted but would output a warning.  In 4.x that syntax would be an error.

2. Internally the code for TXT records would change.

Instead of RecordConfig.TxtStrings, we would just store one big string in RecordConfig.target.  
- `GetTargetField()` would return a single (potentially very long) string (it currently returns a quoted string).
- All existing getters/setters would be adjusted for backwards compatibility.  Some might be marked as depreciated.
- Code documentation would be updated to explain the recommended getters/setters.

### "cons" to this proposal

Maybe we should just leave it alone.  It's the TXT record... how much mucking do we really need to do with it?

Over the years we've put much work into TXT records. They started a simple strings, then we added TXTMulti as a "feature" which meant every provider had to treat TXT records as special cases. Then we unified to the .TxtStrings array.  Recently  we've reworked how PopulateFromString() handles TXT records and provided helper functions.

In other words, when we started we didn't know what we wanted; and we keep changing things.  May we should just be happy with what we have?

**Comment by cafferata** [comment]
I could live with the idea that we can only specify a single string. In practice I have never encountered the need to specify the split point of the string myself. I've also never experienced any problems with a TXT not being successfully validated/processed by a third party.

**Comment by tlimoncelli** [comment]
Fixed with 4.7.3!
